runtime/various_test.go

```diff
@@ -0,0 +1,144 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2020 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package runtime
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/onflow/cadence/runtime/parser2"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/cadence/runtime/sema"
+)
+
+func TestVariousAgainstRegression(t *testing.T) {
```

Comment:
I think it might be better to move each test to the related component (parser/lexer, sema/checker, interpreter). This allows running the component's tests individually and detect regressions in one component earlier in the development. For example, if we break the parser, all parser tests could pass and then only when running the runtime tests we would realize there is a regression. 

For example, the first test case is a parser test case, so it should go into `runtime/parser2/parser_test.go`.
The next five test cases are checker tests, so they should go into `runtime/tests/checker`.
For the test cases where no error is expected, the tests should be "duplicated" in both the checker tests and the interpreter tests, so we can ensure they are successfully checked, but then also successfully interpreted (other successful tests are "duplicated" in a similar way)

Reply:
@turbolent I moved the tests around and restructured them to fit with the other tests in those files.